samples/dotnet/kernel-syntax-examples/Example33_CustomChat.cs

```diff
@@ -0,0 +1,171 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.AI.ChatCompletion;
+using RepoUtils;
+
+/**
+ * The following example shows how to plug use a custom chat completion model.
+ *
+ * This might be useful in a few scenarios, for example:
+ * - You are not using OpenAI or Azure OpenAI models
+ * - You are using OpenAI/Azure OpenAI models but the models are behind a web service with a different API schema
+ * - You want to use a local model
+ */
+public sealed class MyChatCompletionService : IChatCompletion
+{
+    private const string OutputAssistantResult = "Hi I'm your SK Custom Assistant and I'm here to help you to create custom chats like this. :)";
+
+    public ChatHistory CreateNewChat(string instructions = "")
+    {
+        var chatHistory = new ChatHistory();
+
+        if (!string.IsNullOrWhiteSpace(instructions))
+        {
+            chatHistory.AddMessage(ChatHistory.AuthorRoles.System, instructions);
+        }
+
+        return chatHistory;
+    }
+
+    public async Task<string> GenerateMessageAsync(ChatHistory chat, ChatRequestSettings? requestSettings = null, CancellationToken cancellationToken = default)
+    {
+        // Forcing a 2 sec delay (Simulating custom LLM lag)
+        await Task.Delay(2000, cancellationToken);
+
+        return OutputAssistantResult;
+    }
+
+    public async IAsyncEnumerable<string> GenerateMessageStreamAsync(
+        ChatHistory chat,
+        ChatRequestSettings? requestSettings = null,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        var streamedOutput = OutputAssistantResult.Split(' ');
+        foreach (string word in streamedOutput)
+        {
+            await Task.Delay(200, cancellationToken);
+            yield return $"{word} ";
+        }
+    }
+}
+
+// ReSharper disable once InconsistentNaming
+public static class Example33_CustomChat
```

Comment:
what about a more descriptive name?

Reply:
@dluc I'm using here almost the same name you used in the `CustomLLM` maybe I could put all those examples in the CustomLLM or rename it as `CustomLLMChat` and `CustomLLM` -> `CustomLLMCompletion`

Reply:
I think the reasoning is that there could be confusion between bringing your own chat model (your example) and implementing a custom chat solution (copilot chat). I'd agree with either merging into CustomLLM, or renaming to something like CustomChatModel